cryptography/src/bls12381/dkg.rs

```diff
@@ -1422,37 +1422,45 @@ impl<V: Variant, S: Signer> Player<V, S> {
         Some(PlayerAck { sig })
     }
 
-    /// Finalize the player, producing an output, and a share.
+    /// Finalize the player, producing an output, a share, and the number of reveals
+    /// that were used to recover the player's share components.
     ///
     /// This should agree with [`observe`], in terms of `Ok` vs `Err` and the
     /// public output, so long as the logs agree. It's crucial that the players
     /// come to agreement, in some way, on exactly which logs they need to use
     /// for finalize.
     ///
+    /// The returned `usize` indicates how many dealers had to reveal their private
+    /// message because this player didn't acknowledge in time. A value of 0 means
+    /// the player participated fully (sent acks to all selected dealers).
+    ///
     /// This will only ever return [`Error::DkgFailed`].
+    #[allow(clippy::type_complexity)]
     pub fn finalize(
         self,
         logs: BTreeMap<S::PublicKey, DealerLog<V, S::PublicKey>>,
         concurrency: usize,
-    ) -> Result<(Output<V, S::PublicKey>, Share), Error> {
+    ) -> Result<(Output<V, S::PublicKey>, Share, usize), Error> {
```

Comment:
Not sure if it would be useful to return how many shares were revealed (it was useful for the test to ensure recovery but may be unnecessary access)